cmd/object-common.go

```diff
@@ -225,7 +225,7 @@ func cleanupDir(storage StorageAPI, volume, dirPath string) error {
 		if err == errFileNotFound {
 			return nil
 		} else if err != nil { // For any other errors fail.
-			return traceError(err)
```

Comment:
Better to do traceError() at a point where the error happens, because if we decide enhance traceError() to add more context, for ex:

traceError(err, fmt.Sprintf("failed to ListDir on %s %s", volume, entryPath))

it can't be done if we move it to a different place.

what was the reason you moved it down? to avoid stack info for a recursive call?